frontend/public/reducers/features.ts

```diff
@@ -94,14 +94,11 @@ export const featureReducer = (state: FeatureState, action: FeatureAction): Feat
   }
 };
 
-export const stateToFlagsObject = (state: RootState): FlagsObject =>
-  state[featureReducerName].toObject();
-
-export const stateToProps = (desiredFlags: string[], state: RootState): WithFlagsProps => ({
-  flags: desiredFlags.reduce(
-    (allFlags, f) => ({ ...allFlags, [f]: state[featureReducerName].get(f) }),
-    {},
-  ),
+export const stateToFlagsObject = (state: FeatureState, desiredFlags: string[]): FlagsObject =>
+  desiredFlags.reduce((allFlags, f) => ({ ...allFlags, [f]: state.get(f) }), {} as FlagsObject);
```

Comment:
support `desiredFlags` as option so we can continue to use this selector as previously implemented.
If no `desiredFlags` supplied, fallback to all flags.

Reply:
We can do that, but we should also differentiate between "empty flags" (empty array) vs. "all flags" (falsy value) cases.

I'll make `desiredFlags` optional and fallback to `null` which would allow us to reuse it like before :smiley:

Reply:
I didn't realize this function was using `FeatureState`.
This is change request was related to https://github.com/openshift/console/pull/4211#discussion_r388403466 where it would have been nice to keep a selector.